lib/hamster/hash.rb

```diff
@@ -590,6 +590,27 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `hil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return value
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      else
+        if value.respond_to?(:dig)
+          value.dig(*rest)
+        end
+      end
+    end
```

Comment:
I think this implementation could actually be used for vectors too since it is just implemented on top of `[]`.

only reason I didn't solve them both at once is that vectors dig in ruby 2.3.0-dev behaves slightly different.
```ruby
irb(main):006:0> [1,2,3].dig(:first)
TypeError: no implicit conversion of Symbol into Integer
	from (irb):6:in `dig'
	from (irb):6
	from /Users/adam/.rbenv/versions/2.3.0-dev/bin/irb:11:in `<main>'
```
vs.
```ruby
irb(main):007:0> {a: 'b'}.dig(:first)
=> nil
```

with this code, instead of raising an error it would return nil.
However, this behavior might not make it to final release.
https://bugs.ruby-lang.org/issues/11762